### PR TITLE
Support timezones by all types of generators

### PR DESCRIPTION
**Pull request by lptn**
Related issue #23

## Specs
 1. All generators work with different timezones.
 1. All-day events should ignore timezones.

## Changes

In previous implementation `Link::from` and `Link::to` return strings (They also had wrong PHPDoc type).  In order to support tz I have to made some breaking changes : return `\DateTime` objects instead.

I also fixed `allDay` option for Yahoo (found it by debugging their clientside JS).

Additionally I added links to docs (official first, if available) using `@see` PHPDoc block to all Generator classes.

## Drawbacks I know
 - ics: [We should not specify tz for UTC datetime objects](https://icalendar.org/iCalendar-RFC-5545/3-2-19-time-zone-identifier.html), but it's working on google calendar and iCalendar. May be I'll improve it in the future
 - ics: current and previous versions don't pass [validation](https://icalendar.org/validator.html). This is a subject for further improvements

### CODE REVIEW

**Comment by lptn** [comment]
ping @sebastiandedeyne 
Could you please review it?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 All notable changes to `calendar-links` will be documented in this file
 
+## 1.1.1 - 2018-10-08
```
Comment (sebastiandedeyne) [inline CHANGELOG.md]: Please don't modify the changelog in this PR, I'll take care of this when it gets released :)

**Comment by lptn** [comment]
@sebastiandedeyne 
>ics: current and previous versions don't pass validation. This is a subject for further improvements

There are some validators for ics:
 - https://icalendar.org/validator.html
 - http://ical-validator.herokuapp.com/validate/

And content generated by our ICS generator doesn't pass validation due to some minor issues (sorry, I don't remember exactly which ones)

File: CHANGELOG.md (hunk 1 above)
Comment (lptn) [reply CHANGELOG.md]: image
that's not my commit. But maybe I did something wrong on rebasing from upstream

**Comment by lptn** [comment]
Hey
@sebastiandedeyne
@freekmurze 
Could you please review this PR. Other contributors and me waiting for updates from your side on this PR :) (more than 1 month passed after latest changes on this PR)

**Comment by freekmurze** [comment]
Hi @lptn,

we don't have the time anymore to maintain this package. Would you like to step us as the maintainer? 

This is what we expect of a maintainer: You can handle incoming issue and review PRs. You're also allowed to tag released. If you want to make a breaking change and tag a major release please get in contact with us first. Also, if you have questions, feel free to get in touch with us.

If you're up for it great, I'll give you write rights?
It not, no worries, I'll put a "searching for maintainers" message in the readme.

**Comment by lptn** [comment]
Hey @freekmurze 

> Would you like to step us as the maintainer?

Yes, I can. We actively use this package on my company project. I also want to work more on open-source projects, so I have enough motivation to be a maintainer of THIS project. Thanks for this opportunity!

Do you have any ideas (refractor code or tests, add new features, etc) for this package?

Do you have a repository template or a list of best practices for repositories in order to follow your standards? (I assume that on spatie you use some sort of them).

**Comment by freekmurze** [comment]
@lptn That's great to hear, I'll give you write access to this repo.

We have some coding guidelines here: https://guidelines.spatie.be/code-style/laravel-php.

All our packages follow semver, so please don't make any breaking changes unless create a major new version. 

I don't have any major ideas for this package atm, maybe @sebastiandedeyne

**Pull request merged**